### PR TITLE
Extend libtinyiiod

### PR DESCRIPTION
**Pull request by cristiantpop**
Extend parser to support new commands

### CODE REVIEW

**Comment by cristiantpop** [comment]
V1:

- modify commit comments to be more explicit

**Comment by cristiantpop** [comment]
V2: 
- add separate commit for "TIMEOUT" command and set value for it, if the user, provides an implementation for it;
- add separate commit for "WRITEBUF" command; 
- reword commit comments to be more explicit.

**Review by dbogdan — APPROVED** [review]

File: parser.c [hunk 1]
```diff
@@ -126,9 +126,7 @@ static int32_t parse_readbuf_string(struct tinyiiod *iiod, char *str)
 	if (str == ptr || *ptr != '\0' || bytes_count < 0)
```
Comment (mirceacaprioru) [inline parser.c:153]: nitpick: You have a typo in the commit message.  number of bytes and functions instead of function.

**Review by mirceacaprioru — APPROVED** [review]

**Comment by cristiantpop** [comment]
V3: 

- Fix typo in commit message: "Functions will return read/written number of bytes, or error in case of
failure"

**Pull request merged**